### PR TITLE
Vicuna Models checkpoints transfer script

### PR DESCRIPTION
**Pull request by sineeli**
Successfully converted checkpoints from vicuna(torch) to Keras3 compatible, please let me know if any refactoring needed.

image

Thanks

### CODE REVIEW

**Comment by sineeli** [comment]
cc: @mattdangerw

**Review by mattdangerw — COMMENTED** [review]
Looks good! A couple comments...

Next up, you could try uploading these to your individual user on Kaggle, and making a PR that updates our presets file here -> https://github.com/keras-team/keras-nlp/blob/master/keras_nlp/src/models/llama/llama_presets.py

That would give us all a way to test the vicuna models end to end, then we can copy them to the Keras org on Kaggle when they look good.

Thanks!

File: tools/checkpoint_conversion/convert_vicuna_checkpoints.py [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+from keras import ops
+from transformer import LlamaForCausalLM
+from transformers import AutoTokenizer
+
+from keras_nlp import upload_preset
+from keras_nlp.models import LlamaBackbone
+from keras_nlp.models import LlamaCausalLMPreprocessor
+from keras_nlp.models import LlamaTokenizer
+
+PRESET_MAP = {"vicuna_1.5_7b_en": "lmsys/vicuna-7b-v1.5"}
```
[the remark this answers is not in the document]
Comment (sineeli) [reply tools/checkpoint_conversion/convert_vicuna_checkpoints.py:33]: Yes the weights are same llam2 architecture, we can merge with existing script. I will try that. Thanks!

**Comment by sineeli** [comment]
@mattdangerw 

When we run on cpu and use .numpy() at the end of the weights it causes an error: `Bfloat16 scalar not surpported` but when we see `phi3` script there is no such usage and at global level the backend set to `torch` and he is loading the weights directly. 

LLam2 all weights are in torch so and we can use the same approach as `phi3` weights convertion script.

Thanks

**Comment by mattdangerw** [comment]
> When we run on cpu and use .numpy() at the end of the weights it causes an error: Bfloat16 scalar not surpported but when we see phi3 script there is no such usage and at global level the backend set to torch and he is loading the weights directly.

@sineeli I think `keras.ops.convert_to_numpy(x)` would gracefully handle bfloat16, maybe try that?

With what precision are the original pytorch checkpoints stored on disk? If they are at float16, we could just do the same (and store at float16 on disk). The disk format does not mean we need to load at that format.

Anyway, as soon as you push with the comments addressed above we can merge this PR. And keep working on the actual checkpoints we ship.

**Comment by mattdangerw** [comment]
@sineeli can you make your kaggle model public? I'll pull in the script but leave the new preset off for now, we can do that on a follow up PR.

**Comment by sineeli** [comment]
> @sineeli can you make your kaggle model public? I'll pull in the script but leave the new preset off for now, we can do that on a follow up PR.

Sure, waiting for page update. Thank!

**Pull request merged**